src/lcals/FIRST_MIN-Cuda.cpp

```diff
@@ -80,24 +80,31 @@ void FIRST_MIN::runCudaVariantImpl(VariantID vid)
 
        FIRST_MIN_MINLOC_INIT;
 
+       const size_t grid_size = RAJA_DIVIDE_CEILING_INT(iend, block_size);
        MyMinLoc* dminloc;
-       cudaErrchk( cudaMalloc( (void**)&dminloc, sizeof(MyMinLoc) ) );
-       cudaErrchk( cudaMemcpy( dminloc, &mymin, sizeof(MyMinLoc),
-                               cudaMemcpyHostToDevice ) );
+       MyMinLoc mymin_block[grid_size]; //per-block min value
+
+       for (Index_type i=0;i<static_cast<Index_type>(grid_size);i++){
+	       memcpy(&(mymin_block[i]),&mymin, sizeof(MyMinLoc));
+       }
+       cudaErrchk( cudaMalloc( (void**)&dminloc, grid_size * sizeof(MyMinLoc) ) );
+       cudaErrchk( cudaMemcpy( dminloc, mymin_block, grid_size*sizeof(MyMinLoc),
+                               cudaMemcpyHostToDevice ) );      
 
-       const size_t grid_size = RAJA_DIVIDE_CEILING_INT(iend, block_size);
        first_min<block_size><<<grid_size, block_size,
                    sizeof(MyMinLoc)*block_size>>>( x,
                                                    dminloc,
                                                    iend );
-       cudaErrchk( cudaGetLastError() );
+       cudaErrchk( cudaMemcpy( &mymin_block, dminloc, grid_size * sizeof(MyMinLoc),
+                              cudaMemcpyDeviceToHost ) );       
 
-       cudaErrchk( cudaMemcpy( &mymin, dminloc, sizeof(MyMinLoc),
-                               cudaMemcpyDeviceToHost ) );
-       m_minloc = RAJA_MAX(m_minloc, mymin.loc);
+       for (Index_type i=0;i<static_cast<Index_type>(grid_size);i++){
+           m_minloc = RAJA_MAX(m_minloc, (mymin_block[i]).loc);
```

Comment:
Don't we want the min here?

Reply:
Good catch. I missed that.

Reply:
Then I think we need to update the other variants right? Or am I interpreting what the algorithm is incorrectly?

Reply:
OK. I think we should be consistent now.